src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.cs

```diff
@@ -154,12 +154,6 @@ public static MemoryMappedFile CreateFromFile(string path, FileMode mode, string
                 throw new ArgumentException(SR.Argument_EmptyFile);
             }
 
-            if (access == MemoryMappedFileAccess.Read && capacity > fileStream.Length)
-            {
-                CleanupFile(fileStream, existed, path);
-                throw new ArgumentException(SR.Argument_ReadAccessWithLargeCapacity);
-            }
```

Comment:
This is changing the order of exceptions we throw, but I think that's ok.  Just wanted to call out that in theory someone could observe it.

Reply:
@stephentoub do we need to document exception order alterations as breaking changes?

Reply:
The same exception still happens if you get something wrong, it's just if you get multiple things wrong you may not get the same error you used to. I think it's probably below the threshold of what we need to document, i.e. no, but I'm open to other opinions... @ericstj, what do you think?

Reply:
I believe we've done this a few times before and didn't document it, for what it's worth.

Reply:
In desktop we'd avoid this type of change completely.  It doesn't hurt to document it.

Reply:
I think we can preserve the order so I'll do that to decrease risk.